tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_Constraints.cpp

```diff
@@ -137,6 +143,149 @@ void test_gauge_constraint_analytic(
+  const auto& dt_spatial_metric =
+      get<Tags::dt<gr::Tags::SpatialMetric<3>>>(vars);
+  const auto& d_spatial_metric =
+      get<typename Solution::template DerivSpatialMetric<DataVector>>(vars);
+
+  // Compute quantities from variables for the two-index constraint
+  const auto& inverse_spatial_metric =
+      determinant_and_inverse(spatial_metric).second;
+  const auto& inverse_spacetime_metric =
+      gr::inverse_spacetime_metric(lapse, shift, inverse_spatial_metric);
+  const auto& normal_one_form =
+      gr::spacetime_normal_one_form<3, Frame::Inertial>(lapse);
+  const auto& normal_vector = gr::spacetime_normal_vector(lapse, shift);
+
+  // Arbitrary choice for gamma2
+  const auto& gamma2 = make_with_value<Scalar<DataVector>>(x, 4.0);
+
+  // Compute derivatives d_psi, d_phi, d_pi, and d_gauge_function numerically
+  Variables<VariablesTags> gh_vars(data_size);
+  auto& spacetime_metric = get<SpacetimeMetric>(gh_vars);
+  auto& pi = get<Pi>(gh_vars);
+  auto& phi = get<Phi>(gh_vars);
+  auto& gauge_function = get<GaugeH>(gh_vars);
+  spacetime_metric = gr::spacetime_metric(lapse, shift, spatial_metric);
+  phi = GeneralizedHarmonic::phi(lapse, d_lapse, shift, d_shift, spatial_metric,
+                                 d_spatial_metric);
+  pi = GeneralizedHarmonic::pi(lapse, dt_lapse, shift, dt_shift, spatial_metric,
+                               dt_spatial_metric, phi);
+  gauge_function = GeneralizedHarmonic::gauge_source(
+      lapse, dt_lapse, d_lapse, shift, dt_shift, d_shift, spatial_metric,
+      trace(gr::extrinsic_curvature(lapse, shift, d_shift, spatial_metric,
+                                    dt_spatial_metric, d_spatial_metric),
+            inverse_spatial_metric),
+      trace_last_indices(gr::christoffel_first_kind(d_spatial_metric),
+                         inverse_spatial_metric));
+
+  // Compute numerical derivatives of psi,pi,phi,H
+  const auto gh_derivs =
+      partial_derivatives<VariablesTags, VariablesTags, 3, Frame::Inertial>(
+          gh_vars, mesh, coord_map.inv_jacobian(x_logical));
+  const auto& d_spacetime_metric =
+      get<Tags::deriv<SpacetimeMetric, tmpl::size_t<3>, Frame::Inertial>>(
+          gh_derivs);
+  const auto& d_pi =
+      get<Tags::deriv<Pi, tmpl::size_t<3>, Frame::Inertial>>(gh_derivs);
+  const auto& d_phi =
+      get<Tags::deriv<Phi, tmpl::size_t<3>, Frame::Inertial>>(gh_derivs);
+  const auto& d_gauge_function =
+      get<Tags::deriv<GaugeH, tmpl::size_t<3>, Frame::Inertial>>(gh_derivs);
+
+  // Compute the three-index constraint
+  const auto& three_index_constraint =
+      GeneralizedHarmonic::three_index_constraint(d_spacetime_metric, phi);
+
+  // Get the constraint, and check that it vanishes to error_tolerance
+  Approx numerical_approx =
+      Approx::custom().epsilon(error_tolerance).scale(1.0);
+
+  auto two_index_constraint =
+      make_with_value<tnsr::ia<DataVector, 3, Frame::Inertial>>(x, 0.0);
```

Comment:
This should be initialized with `signaling_NaN` to test the initialization to zero in the implementation.